### PR TITLE
feat(alerts): Add new model to represent a snuba query. Change aggregate field to be a string.

### PR DESCRIPTION
**Pull request by wedamija**
As part of performance alerts we need to support custom aggregates. To do so, we're going to use the
same format as discover, which is to store the aggregate as a string, which can look like:
 - `count()`
 - `error_rate()`
 - `avg(transaction.duration)`

At the moment, query fields are duplicated across both `AlertRule` and `SubscriptionQuery`. I'm
going to clean this up with the new `SnubaQuery` model. The data model will look like:

AlertRule M -> 1 Query 1 -> M QuerySubscription

This allows us to clean up our data model a fair bit:
 - `AlertRuleQuerySubscription` will no longer be needed, since we can join through the query.
 - `AlertRuleEnvironment` will no longer be needed. Since we will now only have 0-1
   environments per alert rule, we can just have that reference in `SnubaQuery` and remove it from
   other tables.
 - We can remove the duplicate fields from `AlertRule` and `QuerySubscription`.
 - There will be one source of truth for what a query actually looks like, rather than it being
   duplicated over many `QuerySubscription` rows.

### CODE REVIEW

**Comment by wedamija** [comment]
Generated SQL:
```
BEGIN;
--
-- Create model SnubaQuery
--
CREATE TABLE "sentry_snubaquery" ("id" bigserial NOT NULL PRIMARY KEY, "dataset" text NOT NULL, "query" text NOT NULL, "aggregate" text NOT NULL, "time_window" integer NOT NULL, "resolution" integer NOT NULL, "date_added" timestamp with time zone NOT NULL, "environment_id" bigint NULL);
--
-- Alter field aggregation on alertrule
--
ALTER TABLE "sentry_alertrule" ALTER COLUMN "aggregation" DROP NOT NULL;
--
-- Alter field dataset on alertrule
--
ALTER TABLE "sentry_alertrule" ALTER COLUMN "dataset" DROP NOT NULL;
--
-- Alter field query on alertrule
--
ALTER TABLE "sentry_alertrule" ALTER COLUMN "query" DROP NOT NULL;
--
-- Alter field resolution on alertrule
--
ALTER TABLE "sentry_alertrule" ALTER COLUMN "resolution" DROP NOT NULL;
--
-- Alter field time_window on alertrule
--
ALTER TABLE "sentry_alertrule" ALTER COLUMN "time_window" DROP NOT NULL;
--
-- Alter field aggregation on querysubscription
--
ALTER TABLE "sentry_querysubscription" ALTER COLUMN "aggregation" DROP NOT NULL;
--
-- Alter field dataset on querysubscription
--
ALTER TABLE "sentry_querysubscription" ALTER COLUMN "dataset" DROP NOT NULL;
--
-- Alter field query on querysubscription
--
ALTER TABLE "sentry_querysubscription" ALTER COLUMN "query" DROP NOT NULL;
--
-- Alter field resolution on querysubscription
--
ALTER TABLE "sentry_querysubscription" ALTER COLUMN "resolution" DROP NOT NULL;
--
-- Alter field time_window on querysubscription
--
ALTER TABLE "sentry_querysubscription" ALTER COLUMN "time_window" DROP NOT NULL;
--
-- Add field snuba_query to alertrule
--
ALTER TABLE "sentry_alertrule" ADD COLUMN "snuba_query_id" bigint NULL UNIQUE;
--
-- Add field snuba_query to querysubscription
--
ALTER TABLE "sentry_querysubscription" ADD COLUMN "snuba_query_id" bigint NULL;
CREATE INDEX "sentry_snubaquery_environment_id_47e3f3e3" ON "sentry_snubaquery" ("environment_id");
ALTER TABLE "sentry_alertrule" ADD CONSTRAINT "sentry_alertrule_snuba_query_id_5590368f_fk_sentry_sn" FOREIGN KEY ("snuba_query_id") REFERENCES "sentry_snubaquery" ("id") DEFERRABLE INITIALLY DEFERRED;
CREATE INDEX "sentry_querysubscription_snuba_query_id_860acc5a" ON "sentry_querysubscription" ("snuba_query_id");
ALTER TABLE "sentry_querysubscription" ADD CONSTRAINT "sentry_querysubscrip_snuba_query_id_860acc5a_fk_sentry_sn" FOREIGN KEY ("snuba_query_id") REFERENCES "sentry_snubaquery" ("id") DEFERRABLE INITIALLY DEFERRED;
COMMIT;
```

File: src/sentry/snuba/models.py [hunk 1]
```diff
@@ -38,6 +38,22 @@ class Meta:
         unique_together = (("query_subscription", "environment"),)
 
 
+class SnubaQuery(Model):
```
Comment (markstory) [inline src/sentry/snuba/models.py:41]: Does this new model need to be wired up to deletions?

**Review by markstory — APPROVED** [review]
The list of simplifications with this new model sound awesome :clap:

File: src/sentry/snuba/models.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/snuba/models.py:41]: Hmm, not sure. The only reference from this -> an org is the `Environment`. We might just be able to rely on Django removing these rows when the environment is deleted.

**Pull request merged**